File: benchmark/c/cpw/find_minimum_number_divided_make_number_perfect_square/find_minimum_number_divided_make_number_perfect_square.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int find_minimum_number_divided_make_number_perfect_square ( int n ) ;
int find_minimum_number_divided_make_number_perfect_square ( int n ) {
  int count = 0, ans = 1;
  while ( n % 2 == 0 ) {
    count ++;
    n /= 2;
  }
  if ( count % 2 ) ans *= 2;
  for ( int i = 3;
  i <= sqrt ( n );
  i += 2 ) {
    count = 0;
    while ( n % i == 0 ) {
      count ++;
      n /= i;
    }
    if ( count % 2 ) ans *= i;
  }
  if ( n > 2 ) ans *= n;
  return ans;
}

File: experimental data/RQ3&RQ4/translation_result/fail/c/cpw/mistral/CAPR/few-shot-True/cot-ast/candidates-1/find_minimum_number_divided_make_number_perfect_square/find_minimum_number_divided_make_number_perfect_square.rs


fn find_minimum_number_divided_make_number_perfect_square(n: i32) -> i32 {
    let mut count = 0;
    let mut ans = 1;
    let mut n = n;

    while n % 2 == 0 {
        count += 1;
        n = n.wrapping_shr(1);
    }

    if count % 2 == 1 {
        ans *= 2;
    }

    let mut i = 3;
    while i <= (n as f64).sqrt() as i32 {
        count = 0;
        n = n.wrapping_div(i as i32);

        while n % i == 0 {
            count += 1;
            n = n.wrapping_div(i as i32);
        }

        if count % 2 == 1 {
            ans *= i;
        }

        i += 2;
    

    if n > 2 {
        ans *= n;
    }

    ans
}
